pkg/operator/operator.go

```diff
@@ -293,15 +292,14 @@ func New(logger logr.Logger, clientConfig *rest.Config, opts Options) (*Operator
 // custom resources and registers handlers with the webhook server.
 func (o *Operator) setupAdmissionWebhooks(ctx context.Context) error {
 	// Write provided cert files.
-	caBundle, err := o.ensureCerts(ctx, o.manager.GetWebhookServer().(*webhook.DefaultServer).Options.CertDir)
+	caBundle, err := o.ensureCerts(o.manager.GetWebhookServer().(*webhook.DefaultServer).Options.CertDir)
 	if err != nil {
 		return err
 	}
 
-	if updateAllowed, err := o.canUpdateWebhooks(ctx); err != nil {
```

Comment:
So I proposed removing this step. Rationales:

* It's clear this code is tricky to maintain, we still for some reason check wrongly here, so removing this complexity would be ideal.
* Do we need to support case of NOT injecting periodically caBundle? If yes, perhaps flag would help?

Reply:
I don't know the history of this step. Especially I am not aware of any use case where somebody else would manually set caBundle 🤔

Reply:
So before https://github.com/GoogleCloudPlatform/prometheus-engine/pull/661 we didn't have this check. Sounds like the intention was to have one check instead of failing indefinitely, but did we see ever that happening or harmful @bernot-dev @TheSpiritXIII ? 🤔 

I think it should be safe to have periodic errors in cases of perms error.

Reply:
Your reasoning makes sense. Great investigation! I like how this code looks simpler.

Reply:
More context from @bernot-dev :

> #661 is part of the work necessary to handle the case where certs will be provided by GKE and the bundle will not need to be set.

Let's iterate over that in the next PRs then (with more tests, bit more commentary on the intended behavior). One idea is to maybe decide about this behavior based on some extra annotation/label/operator config instead of permission? The disadvantage of perm based deciding is that we might miss perms misconfiguration cases easily. Anyway, let's iterate!